src/rgw/rgw_lua_utils.h

```diff
@@ -193,5 +193,88 @@ void set_package_path(lua_State* L, const std::string& install_dir);
 // and the "debug" library
 void open_standard_libs(lua_State* L);
 
+typedef int MetaTableClosure(lua_State* L);
+
+template<typename MapType>
+int StringMapWriteableNewIndex(lua_State* L) {
+  const auto map = reinterpret_cast<MapType*>(lua_touserdata(L, lua_upvalueindex(1)));
+
+  const char* index = luaL_checkstring(L, 2);
+  const char* value = luaL_checkstring(L, 3);
+  map->insert_or_assign(index, value);
+  return NO_RETURNVAL;
+}
+
+template<typename MapType=std::map<std::string, std::string>,
```

Comment:
please use your own map type, that has a mutex around the relevant functions
for simplicity, you can wrap it with your class, and expose only the methods used here (when you add the mutex protection)